src/modules/github/components/BulkMergePrs.tsx

```diff
@@ -1,27 +1,109 @@
 'use client'
 
-import type { FC } from 'react'
+import { type FC, useEffect, useState } from 'react'
+
+import { AboutPRButton } from '@/components/github/root/AboutPRButton'
+import { DataList } from '@/components/github/root/DataList'
 
 import { usePrsGroup } from '@/hooks/github/root/usePrsGroup'
 
+import { type TRepoHasCheck } from '@/types/github/root/index'
+
 export const BulkMergePrs: FC = () => {
   const { prsGroup } = usePrsGroup()
+  const [repoHasCheckArray, setRepoHasCheckArray] = useState<TRepoHasCheck[]>(
+    []
+  )
+
+  const handlePullChange = (repo: string, number: number) => {
+    setRepoHasCheckArray((prev) =>
+      prev.map((data) => {
+        if (data.repo === repo) {
+          return {
+            ...data,
+            pulls: data.pulls.map((pull) => {
+              if (pull.number === number) {
+                return { ...pull, isChecked: !pull.isChecked }
+              }
+              return pull
+            })
+          }
+        }
+        return data
+      })
+    )
+  }
+
+  const handlePullCheckAllToggle = (repo: string, isRepoChecked: boolean) => {
+    if (isRepoChecked) {
+      setRepoHasCheckArray((prev) =>
+        prev.map((data) => {
+          if (data.repo === repo) {
+            return {
+              ...data,
+              pulls: data.pulls.map((pull) => ({
+                ...pull,
+                isChecked: false
+              }))
+            }
+          }
+          return data
+        })
+      )
+    } else {
+      setRepoHasCheckArray((prev) =>
+        prev.map((data) => {
+          if (data.repo === repo) {
+            return {
+              ...data,
+              pulls: data.pulls.map((pull) => ({
+                ...pull,
+                isChecked: true
+              }))
+            }
+          }
+          return data
+        })
+      )
+    }
+  }
```

Comment:
It would be better not to pass `isRepoChecked: boolean` parameter and make the same result. Try it.